File: py/countDomains.py

def countDomains(L):
    resultD = dict()
    resultL = []
    for url in L:
        iResults = url.split()
        count = int(iResults[0])
        domain = iResults[1]
        

        while len(domain) > 0:
            if domain in resultD:
                resultD[domain] += count
            else:
                resultD[domain] = count

            nextDotIndex = domain.find('.')
            if nextDotIndex != -1:
                domain = domain[nextDotIndex+1:]
            else:
                break
        # good reminder of how close I was

    for m in resultD: # for each key, where m is the key (in this case it's the actual URL)
        tempStr = str(resultD[m])
        tempStr += ' '
        tempStr += m
        resultL.append(tempStr)

    return resultL


result = countDomains(["900 google.mail.com", "50 yahoo.com", "1 intel.mail.com", "5 wiki.org"])
print(result)

File: js/countDomains.js

function countDomains(cpdomains) {


    resultM = {};
    resultA = [];
    let count = 0;
    let domString = '';
    let domainSplit;
    let tempStr;

    cpdomains.forEach((domain) => {

        domainSplit = domain.split(' ');
        count = parseInt(domainSplit[0]);
        domString = domainSplit[1];

        // while loop here for the real logic
        while (domString.length > 0) {
            if (resultM[domString]) {
                resultM[domString] += count;
            } else {
                resultM[domString] = count;
            }
            if (domString.indexOf('.') == -1) {
                break;
            }
            domString = domString.slice(domString.indexOf('.')+1);
        }

    });

    for (m in resultM) {
        tempStr = resultM[m];
        tempStr += ' ';
        tempStr += m;
        resultA.push(tempStr);
    }
    return resultA;

}



const result = countDomains(["900 google.mail.com", "50 yahoo.com", "1 intel.mail.com", "5 wiki.org"]);
console.log(result);